core/test/XiosReadWrite_test.cpp

```diff
@@ -0,0 +1,318 @@
+/*!
+ * @file    XiosReadWrite_test.cpp
+ * @author  Joe Wallwork <jw2423@cam.ac.uk>
+ * @date    19 Nov 2024
+ * @brief   Tests for XIOS write method
+ * @details
+ * This test is designed to test the read and write methods of the C++
+ * interface for XIOS.
+ *
+ */
+#include <doctest/extensions/doctest_mpi.h>
+#undef INFO
+
+#include "StructureModule/include/ParametricGrid.hpp"
+#include "include/NextsimModule.hpp"
+#include "include/ParaGridIO.hpp"
+#include "include/Xios.hpp"
+
+#include <filesystem>
+
+namespace Nextsim {
+
+/*!
+ * Create a formatted attribute ID.
+ *
+ * @param label Label to use for the attribute format
+ * @param dim Spatial dimensions
+ * @return "<label>_<dim>D"
+ */
+std::string formatId(const std::string label, const int dim)
+{
+    return (boost::format("%1$s_%2$1dD") % label % dim).str();
+}
+
+/*!
+ * Set up the XIOS handler class for testing file reading and writing.
+ *
+ * The function assumes two MPI ranks.
+ *
+ * @param dim The number of spatial dimensions
+ * @param read If true, set up for file reading test, otherwise for file writing test
+ * @return Appropriately configured Xios handler class instance
+ */
+Xios setupXiosHandler(int dim, bool read)
+{
+    if ((dim != 2) && (dim != 3)) {
+        throw std::invalid_argument("Test only implemented for 2D and 3D cases");
+    }
+
+    // Create ParametricGrid and ParaGridIO instances
+    Module::setImplementation<IStructure>("Nextsim::ParametricGrid");
+    ParametricGrid grid;
+    ParaGridIO* pio = new ParaGridIO(grid);
+    grid.setIO(pio);
+
+    // Initialize an Xios instance called xios_handler
+    // TODO: Create XIOS handler along with ParaGridIO instance
```

Comment:
As in the previous test scripts, these TODOs are indicators for the design process for closer integration of XIOS.